### ISSUE TITLE
[SEMANTIC MARKUP] Checkbox Component - Extend the aria-describedby attribute to accept more than one ID

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#26988]
## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Details

The Design System team has been [working on an Attestation component](https://github.com/department-of-veterans-affairs/vets-website/pull/17525) that extends our FormSignature. We want to add an `aria-describedby` attribute that will read out the attestation statement when screen reader users check the box. This gives users a second chance to hear what they have attested (spouse, children, etc.).

Our Checkbox component only accepts one prop for `errorSpanId`. I'd like to add a prop `descriptionId` that isn't required but will cause screen readers to announce the text that is referenced by `descriptionId` when users check the box.

## Acceptance Criteria

- [ ] `descriptionId` is not a required prop
- [ ] `descriptionId` and `errorSpanId` can both be present, one present, or neither present, and announce properly to screen readers
- [ ] axe-core shows 0 violations

## Environment

* https://github.com/department-of-veterans-affairs/component-library/blob/master/src/components/Checkbox/Checkbox.jsx

## Proposed Solution (if known)

```diff
! Checkbox#L69

<input
  aria-labelledby={this.props.ariaLabelledBy}
- aria-describedby={errorSpanId}
+ aria-describedby={ `${errorSpanId} ${this.props.descriptionId}` }
  checked={this.props.checked}
  id={this.inputId}
  name={this.props.name}
  type="checkbox"
  onChange={this.handleChange}
/>
```

**Comment by caw310** [comment]
DS workshop: Group 1 - Medium priority

**Comment by bkjohnson** [comment]
Closing this since it is specific to the deprecated `<Checkbox>` React component. The web component with Shadow DOM cannot reference any IDs external to the component.

We could consider making a change so that the [`aria-describedby` would point to the `description` id](https://github.com/department-of-veterans-affairs/component-library/blob/ff6baa66f08b809d2bf3c4276c3da4f034b3968e/packages/web-components/src/components/va-checkbox/va-checkbox.tsx#L116-L130) as well as `error-message`.

cc @k80bowman
